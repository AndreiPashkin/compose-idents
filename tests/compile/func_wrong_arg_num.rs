use compose_idents::compose_idents;

compose_idents!(my_fn = [snake_case(fooBar, barBaz)], {
    fn my_fn() -> u32 {
        42
    }
});

fn main() {
    assert_eq!(foo_baz(), 42);
}
