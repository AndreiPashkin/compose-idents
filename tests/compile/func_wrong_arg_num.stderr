error: Type constraints for function "snake_case(arg)" are not satisfied
 --> tests/compile/func_wrong_arg_num.rs:3:25
  |
3 | compose_idents!(my_fn = snake_case(fooBar, barBaz), {
  |                         ^^^^^^^^^^

error[E0425]: cannot find function `foo_baz` in this scope
  --> tests/compile/func_wrong_arg_num.rs:10:16
   |
10 |     assert_eq!(foo_baz(), 42);
   |                ^^^^^^^ not found in this scope
